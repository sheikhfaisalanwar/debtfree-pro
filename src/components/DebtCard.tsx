import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { Debt, DebtType } from '../types/Debt';

interface DebtCardProps {
  debt: Debt;
  isPriority?: boolean;
  onPress?: () => void;
}

export const DebtCard: React.FC<DebtCardProps> = ({ debt, isPriority, onPress }) => {
  const getDebtTypeIcon = (type: DebtType): string => {
    switch (type) {
      case DebtType.CREDIT_CARD: return '💳';
      case DebtType.AUTO_LOAN: return '🚗';
      case DebtType.PERSONAL_LOAN: return '💰';
      case DebtType.LINE_OF_CREDIT: return '📈';
      case DebtType.STUDENT_LOAN: return '🎓';
      case DebtType.MORTGAGE: return '🏠';
      default: return '📄';
    }
  };

  const formatCurrency = (amount: number): string => {
    return new Intl.NumberFormat('en-CA', {
      style: 'currency',
      currency: 'CAD',
    }).format(amount);
  };

  return (
    <TouchableOpacity
      style={[styles.container, isPriority && styles.priorityContainer]}
      onPress={onPress}
    >
      <View style={styles.header}>
        <View style={styles.titleRow}>
          <Text style={styles.icon}>{getDebtTypeIcon(debt.type)}</Text>
          <Text style={styles.name}>{debt.name}</Text>
          {isPriority && <View style={styles.priorityBadge}>
            <Text style={styles.priorityText}>FOCUS</Text>
          </View>}
        </View>
        <Text style={styles.balance}>{formatCurrency(debt.balance)}</Text>
      </View>
      
      <View style={styles.details}>
        <Text style={styles.detailText}>
          {debt.interestRate.toFixed(2)}% APR • Min: {formatCurrency(debt.minimumPayment)}/month
        </Text>
        {debt.institution && (
          <Text style={styles.institution}>{debt.institution}</Text>
        )}
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderRadius: 12,
    padding: 16,
    marginBottom: 12,
    borderWidth: 1,
    borderColor: '#e5e7eb',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  priorityContainer: {
    borderColor: '#ef4444',
    backgroundColor: 'rgba(239, 68, 68, 0.05)',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'flex-start',
    marginBottom: 8,
  },
  titleRow: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
  },
  icon: {
    fontSize: 20,
    marginRight: 8,
  },
  name: {
    fontSize: 16,
    fontWeight: '600',
    color: '#1f2937',
    flex: 1,
  },
  balance: {
    fontSize: 18,
    fontWeight: '700',
    color: '#1f2937',
  },
  priorityBadge: {
    backgroundColor: '#ef4444',
    paddingHorizontal: 8,
    paddingVertical: 4,
    borderRadius: 12,
    marginLeft: 8,
  },
  priorityText: {
    color: 'white',
    fontSize: 10,
    fontWeight: '600',
  },
  details: {
    marginTop: 4,
  },
  detailText: {
    fontSize: 14,
    color: '#6b7280',
  },
  institution: {
    fontSize: 12,
    color: '#9ca3af',
    marginTop: 2,
  },
});