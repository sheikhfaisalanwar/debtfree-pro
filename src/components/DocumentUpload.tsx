import React, { useState } from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
  Alert,
  ActivityIndicator,
} from 'react-native';
import { StatementProcessingService } from '../services/StatementProcessingService';
import { DocumentManagerService } from '../services/DocumentManagerService';
import { ManualDebtEntryModal } from './ManualDebtEntryModal';
import { Debt } from '../types/Debt';

interface DocumentUploadProps {
  debtId?: string;
  onUploadSuccess?: () => void;
  onUploadError?: (error: string) => void;
}

export const DocumentUpload: React.FC<DocumentUploadProps> = ({
  debtId,
  onUploadSuccess,
  onUploadError,
}) => {
  const [uploading, setUploading] = useState(false);
  const [lastUploadResult, setLastUploadResult] = useState<string | null>(null);
  const [showManualEntryModal, setShowManualEntryModal] = useState(false);
  const [pendingStatement, setPendingStatement] = useState<any>(null);
  const [createdDebtId, setCreatedDebtId] = useState<string | null>(null);

  const handleDocumentUpload = async () => {
    try {
      setUploading(true);
      setLastUploadResult(null);

      // First, upload and process the document to see what data we get
      const uploadResult = await DocumentManagerService.uploadAndProcessDocument(debtId);
      
      if (!uploadResult.success || !uploadResult.statement) {
        throw new Error(uploadResult.error || 'Upload failed');
      }

      // Check if meaningful data was extracted from the PDF
      const hasExtractedData = (uploadResult.statement as any).hasExtractedData;
      
      if (hasExtractedData && debtId) {
        // PDF data was extracted and we have a debtId - proceed with automatic processing
        const result = await StatementProcessingService.processUploadedStatement(debtId);

        if (result.updated && result.statement && result.analysis) {
          const analysisText = StatementProcessingService.formatAnalysisSummary(result.analysis);
          setLastUploadResult(`✅ Upload successful! ${analysisText}`);
          
          Alert.alert(
            'Upload Successful! 🎉',
            `Statement processed successfully.\n\n${analysisText}`,
            [{ text: 'OK', style: 'default' }]
          );
          
          onUploadSuccess?.();
        } else if (result.statement) {
          const validationSummary = result.statement.fileName 
            ? `Statement "${result.statement.fileName}" uploaded successfully`
            : 'Statement uploaded successfully';
          
          setLastUploadResult(`✅ ${validationSummary}`);
          Alert.alert('Upload Complete', validationSummary);
          onUploadSuccess?.();
        } else {
          throw new Error(result.error || 'Processing failed');
        }
      } else {
        // No meaningful data extracted or no debtId - show manual entry modal
        setPendingStatement(uploadResult.statement);
        setShowManualEntryModal(true);
        setLastUploadResult(`📄 Document uploaded. Please enter debt details manually.`);
      }
    } catch (error) {
      const errorMessage = error instanceof Error ? error.message : 'Upload failed';
      setLastUploadResult(`❌ ${errorMessage}`);
      
      Alert.alert(
        'Upload Failed',
        errorMessage,
        [{ text: 'OK', style: 'default' }]
      );
      
      onUploadError?.(errorMessage);
    } finally {
      setUploading(false);
    }
  };

  const handleDebtCreated = async (debt: Debt) => {
    try {
      // Store the created debt ID in temporary state
      setCreatedDebtId(debt.id);
      
      if (pendingStatement) {
        // Process the existing statement with the newly created debt ID
        const result = await StatementProcessingService.processExistingStatement(pendingStatement.id, debt.id);
        
        if (result.statement) {
          setLastUploadResult(`✅ Debt created and statement linked successfully! (Debt ID: ${debt.id})`);
          Alert.alert(
            'Success! 🎉',
            `Debt "${debt.name}" has been created with ID ${debt.id} and your statement has been linked to it.`,
            [{ text: 'OK', style: 'default' }]
          );
          onUploadSuccess?.();
        } else {
          throw new Error(result.error || 'Failed to process statement');
        }
      }
    } catch (error) {
      Alert.alert(
        'Error',
        `Failed to link statement to debt: ${error instanceof Error ? error.message : 'Unknown error'}`,
        [{ text: 'OK', style: 'default' }]
      );
    } finally {
      // Clear temporary states
      setPendingStatement(null);
      setCreatedDebtId(null);
    }
  };

  const handleModalClose = () => {
    setShowManualEntryModal(false);
    setPendingStatement(null);
    setCreatedDebtId(null);
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>📄 Upload Statement</Text>
        <Text style={styles.subtitle}>
          Upload CSV or PDF statements to automatically update your debt information
        </Text>
      </View>

      <TouchableOpacity
        style={[styles.uploadButton, uploading && styles.uploadButtonDisabled]}
        onPress={handleDocumentUpload}
        disabled={uploading}
        activeOpacity={0.7}
      >
        {uploading ? (
          <View style={styles.uploadingContent}>
            <ActivityIndicator size="small" color="#ffffff" />
            <Text style={styles.uploadButtonText}>Processing...</Text>
          </View>
        ) : (
          <View style={styles.uploadContent}>
            <Text style={styles.uploadIcon}>📎</Text>
            <Text style={styles.uploadButtonText}>Choose Document</Text>
            <Text style={styles.uploadHint}>CSV or PDF</Text>
          </View>
        )}
      </TouchableOpacity>

      {lastUploadResult && (
        <View style={styles.resultContainer}>
          <Text style={styles.resultText}>{lastUploadResult}</Text>
        </View>
      )}

      <View style={styles.infoContainer}>
        <Text style={styles.infoTitle}>📋 Supported Formats:</Text>
        <Text style={styles.infoText}>• CSV files with transaction data</Text>
        <Text style={styles.infoText}>• PDF bank and credit card statements</Text>
        <Text style={styles.infoText}>• Maximum file size: 10MB</Text>
      </View>

      <ManualDebtEntryModal
        visible={showManualEntryModal}
        onClose={handleModalClose}
        onDebtCreated={handleDebtCreated}
        fileName={pendingStatement?.fileName}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    borderRadius: 16,
    padding: 20,
    marginHorizontal: 16,
    marginVertical: 8,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 5,
  },
  header: {
    marginBottom: 20,
  },
  title: {
    fontSize: 20,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 8,
  },
  subtitle: {
    fontSize: 14,
    color: '#6b7280',
    lineHeight: 20,
  },
  uploadButton: {
    backgroundColor: '#3b82f6',
    borderRadius: 12,
    padding: 20,
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 16,
    minHeight: 80,
  },
  uploadButtonDisabled: {
    backgroundColor: '#9ca3af',
  },
  uploadContent: {
    alignItems: 'center',
  },
  uploadingContent: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  uploadIcon: {
    fontSize: 24,
    marginBottom: 8,
  },
  uploadButtonText: {
    color: '#ffffff',
    fontSize: 16,
    fontWeight: '600',
    marginLeft: 8,
  },
  uploadHint: {
    color: '#bfdbfe',
    fontSize: 12,
    marginTop: 4,
  },
  resultContainer: {
    backgroundColor: '#f3f4f6',
    borderRadius: 8,
    padding: 12,
    marginBottom: 16,
  },
  resultText: {
    fontSize: 14,
    color: '#374151',
    textAlign: 'center',
  },
  infoContainer: {
    borderTopWidth: 1,
    borderTopColor: '#e5e7eb',
    paddingTop: 16,
  },
  infoTitle: {
    fontSize: 14,
    fontWeight: '600',
    color: '#374151',
    marginBottom: 8,
  },
  infoText: {
    fontSize: 12,
    color: '#6b7280',
    marginBottom: 4,
    lineHeight: 16,
  },
});