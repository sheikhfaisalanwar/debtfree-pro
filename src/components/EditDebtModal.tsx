import React, { useState, useEffect } from 'react';
import {
  View,
  Text,
  Modal,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  Alert,
  ScrollView,
  KeyboardAvoidingView,
  Platform,
} from 'react-native';
import { Debt, DebtType } from '../types/Debt';
import { DataStoreService } from '../services/DataStoreService';

interface EditDebtModalProps {
  visible: boolean;
  debt: Debt;
  onSave: () => void;
  onCancel: () => void;
}

export const EditDebtModal: React.FC<EditDebtModalProps> = ({
  visible,
  debt,
  onSave,
  onCancel,
}) => {
  const [name, setName] = useState('');
  const [type, setType] = useState<DebtType>(DebtType.CREDIT_CARD);
  const [balance, setBalance] = useState('');
  const [minimumPayment, setMinimumPayment] = useState('');
  const [interestRate, setInterestRate] = useState('');
  const [institution, setInstitution] = useState('');
  const [accountNumber, setAccountNumber] = useState('');
  const [dueDate, setDueDate] = useState('');
  const [loading, setLoading] = useState(false);

  // Populate form when debt changes
  useEffect(() => {
    if (debt) {
      setName(debt.name);
      setType(debt.type);
      setBalance(debt.balance.toString());
      setMinimumPayment(debt.minimumPayment.toString());
      setInterestRate(debt.interestRate.toString());
      setInstitution(debt.institution || '');
      setAccountNumber(debt.accountNumber || '');
      setDueDate(debt.dueDate ? debt.dueDate.toString() : '');
    }
  }, [debt]);

  const resetForm = () => {
    setName('');
    setType(DebtType.CREDIT_CARD);
    setBalance('');
    setMinimumPayment('');
    setInterestRate('');
    setInstitution('');
    setAccountNumber('');
    setDueDate('');
  };

  const handleClose = () => {
    resetForm();
    onCancel();
  };

  const validateForm = (): string | null => {
    if (!name.trim()) return 'Debt name is required';
    if (!balance.trim()) return 'Balance is required';
    if (!minimumPayment.trim()) return 'Minimum payment is required';
    if (!interestRate.trim()) return 'Interest rate is required';

    const balanceNum = parseFloat(balance);
    const minPaymentNum = parseFloat(minimumPayment);
    const interestRateNum = parseFloat(interestRate);
    const dueDateNum = dueDate ? parseInt(dueDate) : undefined;

    if (isNaN(balanceNum) || balanceNum < 0) return 'Balance must be a valid positive number';
    if (isNaN(minPaymentNum) || minPaymentNum < 0) return 'Minimum payment must be a valid positive number';
    if (isNaN(interestRateNum) || interestRateNum < 0 || interestRateNum > 100) return 'Interest rate must be between 0 and 100';
    if (dueDateNum && (dueDateNum < 1 || dueDateNum > 31)) return 'Due date must be between 1 and 31';

    return null;
  };

  const handleSave = async () => {
    if (!debt) return;

    const validationError = validateForm();
    if (validationError) {
      Alert.alert('Validation Error', validationError);
      return;
    }

    try {
      setLoading(true);

      const updateParams = {
        id: debt.id,
        name: name.trim(),
        type,
        balance: parseFloat(balance),
        minimumPayment: parseFloat(minimumPayment),
        interestRate: parseFloat(interestRate),
        institution: institution.trim() || undefined,
        accountNumber: accountNumber.trim() || undefined,
        dueDate: dueDate ? parseInt(dueDate) : undefined,
      };

      const updatedDebt = await DataStoreService.updateDebt(updateParams);
      
      if (!updatedDebt) {
        throw new Error('Failed to update debt - debt not found');
      }

      onSave();
      handleClose();

      Alert.alert(
        'Debt Updated Successfully',
        `"${updatedDebt.name}" has been updated successfully.`,
        [{ text: 'OK', style: 'default' }]
      );
    } catch (error) {
      Alert.alert(
        'Error',
        `Failed to update debt: ${error instanceof Error ? error.message : 'Unknown error'}`,
        [{ text: 'OK', style: 'default' }]
      );
    } finally {
      setLoading(false);
    }
  };

  const handleDelete = () => {
    if (!debt) return;

    Alert.alert(
      'Delete Debt',
      `Are you sure you want to delete "${debt.name}"? This action cannot be undone and will also delete all associated statements.`,
      [
        { text: 'Cancel', style: 'cancel' },
        {
          text: 'Delete',
          style: 'destructive',
          onPress: async () => {
            try {
              setLoading(true);
              const success = await DataStoreService.deleteDebt(debt.id);
              
              if (success) {
                Alert.alert(
                  'Debt Deleted',
                  `"${debt.name}" has been deleted successfully.`,
                  [{ text: 'OK', style: 'default' }]
                );
                onSave();
                handleClose();
              } else {
                throw new Error('Failed to delete debt');
              }
            } catch (error) {
              Alert.alert(
                'Error',
                `Failed to delete debt: ${error instanceof Error ? error.message : 'Unknown error'}`,
                [{ text: 'OK', style: 'default' }]
              );
            } finally {
              setLoading(false);
            }
          },
        },
      ]
    );
  };

  const debtTypes = [
    { value: DebtType.CREDIT_CARD, label: 'Credit Card' },
    { value: DebtType.LINE_OF_CREDIT, label: 'Line of Credit' },
    { value: DebtType.AUTO_LOAN, label: 'Auto Loan' },
    { value: DebtType.PERSONAL_LOAN, label: 'Personal Loan' },
    { value: DebtType.STUDENT_LOAN, label: 'Student Loan' },
    { value: DebtType.MORTGAGE, label: 'Mortgage' },
    { value: DebtType.OTHER, label: 'Other' },
  ];

  if (!debt || !visible) return null;

  return (
    <Modal
      visible={visible}
      animationType="slide"
      presentationStyle="pageSheet"
      onRequestClose={handleClose}
    >
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={styles.container}
      >
        <View style={styles.header}>
          <TouchableOpacity onPress={handleClose} style={styles.cancelButton}>
            <Text style={styles.cancelButtonText}>Cancel</Text>
          </TouchableOpacity>
          <Text style={styles.title}>Edit Debt</Text>
          <TouchableOpacity
            onPress={handleSave}
            style={[styles.saveButton, loading && styles.saveButtonDisabled]}
            disabled={loading}
          >
            <Text style={styles.saveButtonText}>{loading ? 'Saving...' : 'Save'}</Text>
          </TouchableOpacity>
        </View>

        <ScrollView style={styles.content} showsVerticalScrollIndicator={false}>
          <View style={styles.infoBox}>
            <Text style={styles.infoTitle}>üìù Editing: {debt.name}</Text>
            <Text style={styles.infoText}>
              Update your debt information to keep track of balance changes and payment progress.
              This is useful for monthly statement updates.
            </Text>
          </View>

          <View style={styles.section}>
            <Text style={styles.sectionTitle}>Basic Information</Text>
            
            <View style={styles.field}>
              <Text style={styles.label}>Debt Name *</Text>
              <TextInput
                style={styles.textInput}
                value={name}
                onChangeText={setName}
                placeholder="e.g., Chase Freedom Card, Car Loan"
                placeholderTextColor="#9ca3af"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Debt Type *</Text>
              <ScrollView horizontal showsHorizontalScrollIndicator={false} style={styles.typeContainer}>
                {debtTypes.map((debtType) => (
                  <TouchableOpacity
                    key={debtType.value}
                    style={[
                      styles.typeButton,
                      type === debtType.value && styles.typeButtonSelected,
                    ]}
                    onPress={() => setType(debtType.value)}
                  >
                    <Text
                      style={[
                        styles.typeButtonText,
                        type === debtType.value && styles.typeButtonTextSelected,
                      ]}
                    >
                      {debtType.label}
                    </Text>
                  </TouchableOpacity>
                ))}
              </ScrollView>
            </View>
          </View>

          <View style={styles.section}>
            <Text style={styles.sectionTitle}>Financial Details</Text>
            
            <View style={styles.field}>
              <Text style={styles.label}>Current Balance *</Text>
              <TextInput
                style={styles.textInput}
                value={balance}
                onChangeText={setBalance}
                placeholder="0.00"
                placeholderTextColor="#9ca3af"
                keyboardType="decimal-pad"
              />
              <Text style={styles.fieldHint}>
                Previous: ${debt.balance.toFixed(2)}
              </Text>
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Minimum Payment *</Text>
              <TextInput
                style={styles.textInput}
                value={minimumPayment}
                onChangeText={setMinimumPayment}
                placeholder="0.00"
                placeholderTextColor="#9ca3af"
                keyboardType="decimal-pad"
              />
              <Text style={styles.fieldHint}>
                Previous: ${debt.minimumPayment.toFixed(2)}
              </Text>
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Interest Rate (%) *</Text>
              <TextInput
                style={styles.textInput}
                value={interestRate}
                onChangeText={setInterestRate}
                placeholder="0.00"
                placeholderTextColor="#9ca3af"
                keyboardType="decimal-pad"
              />
              <Text style={styles.fieldHint}>
                Previous: {debt.interestRate.toFixed(2)}%
              </Text>
            </View>
          </View>

          <View style={styles.section}>
            <Text style={styles.sectionTitle}>Additional Information</Text>
            
            <View style={styles.field}>
              <Text style={styles.label}>Institution</Text>
              <TextInput
                style={styles.textInput}
                value={institution}
                onChangeText={setInstitution}
                placeholder="e.g., Chase, Wells Fargo"
                placeholderTextColor="#9ca3af"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Account Number (Last 4 digits)</Text>
              <TextInput
                style={styles.textInput}
                value={accountNumber}
                onChangeText={setAccountNumber}
                placeholder="****"
                placeholderTextColor="#9ca3af"
                maxLength={4}
                keyboardType="number-pad"
              />
            </View>

            <View style={styles.field}>
              <Text style={styles.label}>Due Date (Day of Month)</Text>
              <TextInput
                style={styles.textInput}
                value={dueDate}
                onChangeText={setDueDate}
                placeholder="15"
                placeholderTextColor="#9ca3af"
                keyboardType="number-pad"
                maxLength={2}
              />
            </View>
          </View>

          <View style={styles.dangerSection}>
            <Text style={styles.dangerSectionTitle}>Danger Zone</Text>
            <TouchableOpacity
              style={[styles.deleteButton, loading && styles.deleteButtonDisabled]}
              onPress={handleDelete}
              disabled={loading}
            >
              <Text style={styles.deleteButtonText}>üóëÔ∏è Delete Debt</Text>
            </TouchableOpacity>
            <Text style={styles.dangerText}>
              Deleting this debt will remove it and all associated statements permanently.
            </Text>
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
    </Modal>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f9fafb',
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 20,
    paddingVertical: 16,
    backgroundColor: 'white',
    borderBottomWidth: 1,
    borderBottomColor: '#e5e7eb',
  },
  cancelButton: {
    paddingVertical: 8,
    paddingHorizontal: 12,
  },
  cancelButtonText: {
    fontSize: 16,
    color: '#6b7280',
  },
  title: {
    fontSize: 18,
    fontWeight: '600',
    color: '#1f2937',
  },
  saveButton: {
    backgroundColor: '#3b82f6',
    paddingVertical: 8,
    paddingHorizontal: 16,
    borderRadius: 8,
  },
  saveButtonDisabled: {
    backgroundColor: '#9ca3af',
  },
  saveButtonText: {
    fontSize: 16,
    fontWeight: '600',
    color: 'white',
  },
  content: {
    flex: 1,
    paddingHorizontal: 20,
  },
  infoBox: {
    backgroundColor: '#dbeafe',
    borderRadius: 12,
    padding: 16,
    marginVertical: 20,
    borderLeftWidth: 4,
    borderLeftColor: '#3b82f6',
  },
  infoTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: '#1e40af',
    marginBottom: 8,
  },
  infoText: {
    fontSize: 14,
    color: '#1e40af',
    lineHeight: 20,
  },
  section: {
    backgroundColor: 'white',
    borderRadius: 12,
    padding: 20,
    marginVertical: 12,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 16,
  },
  field: {
    marginBottom: 20,
  },
  label: {
    fontSize: 14,
    fontWeight: '500',
    color: '#374151',
    marginBottom: 8,
  },
  textInput: {
    borderWidth: 1,
    borderColor: '#d1d5db',
    borderRadius: 8,
    paddingHorizontal: 12,
    paddingVertical: 12,
    fontSize: 16,
    color: '#1f2937',
    backgroundColor: 'white',
  },
  fieldHint: {
    fontSize: 12,
    color: '#6b7280',
    marginTop: 4,
    fontStyle: 'italic',
  },
  typeContainer: {
    flexDirection: 'row',
  },
  typeButton: {
    backgroundColor: '#f3f4f6',
    borderRadius: 20,
    paddingHorizontal: 16,
    paddingVertical: 8,
    marginRight: 8,
    borderWidth: 1,
    borderColor: '#e5e7eb',
  },
  typeButtonSelected: {
    backgroundColor: '#3b82f6',
    borderColor: '#3b82f6',
  },
  typeButtonText: {
    fontSize: 14,
    color: '#6b7280',
    fontWeight: '500',
  },
  typeButtonTextSelected: {
    color: 'white',
  },
  dangerSection: {
    backgroundColor: '#fef2f2',
    borderRadius: 12,
    padding: 20,
    marginVertical: 12,
    borderWidth: 1,
    borderColor: '#fecaca',
  },
  dangerSectionTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#dc2626',
    marginBottom: 16,
  },
  deleteButton: {
    backgroundColor: '#dc2626',
    borderRadius: 8,
    paddingVertical: 12,
    paddingHorizontal: 16,
    alignItems: 'center',
    marginBottom: 12,
  },
  deleteButtonDisabled: {
    backgroundColor: '#9ca3af',
  },
  deleteButtonText: {
    fontSize: 16,
    fontWeight: '600',
    color: 'white',
  },
  dangerText: {
    fontSize: 12,
    color: '#dc2626',
    textAlign: 'center',
    lineHeight: 16,
  },
});