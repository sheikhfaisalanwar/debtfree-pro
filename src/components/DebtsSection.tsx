import React from 'react';
import {
  View,
  Text,
  StyleSheet,
} from 'react-native';
import { DebtCard } from './DebtCard';
import { DebtAccountBalance } from '../types/DebtBalance';

interface DebtsSectionProps {
  accountsWithBalances: DebtAccountBalance[];
  onDebtPress: (accountBalance: DebtAccountBalance) => void;
}

export const DebtsSection: React.FC<DebtsSectionProps> = ({
  accountsWithBalances,
  onDebtPress,
}) => {
  return (
    <View style={styles.card}>
      <Text style={styles.cardTitle}>ðŸ’³ Your Debts</Text>
      {accountsWithBalances.length === 0 ? (
        <View style={styles.emptyState}>
          <Text style={styles.emptyStateIcon}>ðŸ’³</Text>
          <Text style={styles.emptyStateTitle}>No debts added yet</Text>
          <Text style={styles.emptyStateText}>
            Use the "Add Debt" button above to get started with your debt payoff journey
          </Text>
        </View>
      ) : (
        accountsWithBalances.map(({ account, balance }, index) => (
          <DebtCard
            key={account.id}
            account={account}
            balance={balance}
            isPriority={index === 0} // First debt in snowball method
            onPress={() => onDebtPress({ account, balance })}
          />
        ))
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: 'white',
    margin: 16,
    marginTop: 0,
    padding: 20,
    borderRadius: 16,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 5,
  },
  cardTitle: {
    fontSize: 20,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 16,
  },
  emptyState: {
    alignItems: 'center',
    paddingVertical: 32,
    paddingHorizontal: 16,
  },
  emptyStateIcon: {
    fontSize: 48,
    marginBottom: 16,
  },
  emptyStateTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#1f2937',
    marginBottom: 8,
    textAlign: 'center',
  },
  emptyStateText: {
    fontSize: 14,
    color: '#6b7280',
    textAlign: 'center',
    lineHeight: 20,
  },
});