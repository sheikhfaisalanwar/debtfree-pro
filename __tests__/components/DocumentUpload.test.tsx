import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { DocumentUpload } from '../../src/components/DocumentUpload';

describe('DocumentUpload', () => {
  it('renders correctly with default state', async () => {
    let tree: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(() => {
      tree = ReactTestRenderer.create(<DocumentUpload />);
    });

    const json = tree!.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with debtId prop', async () => {
    let tree: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(() => {
      tree = ReactTestRenderer.create(<DocumentUpload debtId="debt_123" />);
    });

    const json = tree!.toJSON();
    expect(json).toBeTruthy();
  });

  it('renders with callback props', async () => {
    const mockOnSuccess = jest.fn();
    const mockOnError = jest.fn();
    let tree: ReactTestRenderer.ReactTestRenderer;
    
    await ReactTestRenderer.act(() => {
      tree = ReactTestRenderer.create(
        <DocumentUpload 
          onUploadSuccess={mockOnSuccess} 
          onUploadError={mockOnError} 
        />
      );
    });

    const json = tree!.toJSON();
    expect(json).toBeTruthy();
  });
});