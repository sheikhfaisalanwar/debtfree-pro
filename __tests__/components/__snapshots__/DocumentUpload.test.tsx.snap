// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentUpload renders correctly with default state 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 16,
      "elevation": 5,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      {
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#1f2937",
          "fontSize": 20,
          "fontWeight": "600",
          "marginBottom": 8,
        }
      }
    >
      ðŸ“„ Upload Statement
    </Text>
    <Text
      style={
        {
          "color": "#6b7280",
          "fontSize": 14,
          "lineHeight": 20,
        }
      }
    >
      Upload CSV or PDF statements to automatically update your debt information
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#3b82f6",
        "borderRadius": 12,
        "justifyContent": "center",
        "marginBottom": 16,
        "minHeight": 80,
        "opacity": 1,
        "padding": 20,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 24,
            "marginBottom": 8,
          }
        }
      >
        ðŸ“Ž
      </Text>
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "600",
            "marginLeft": 8,
          }
        }
      >
        Choose Document
      </Text>
      <Text
        style={
          {
            "color": "#bfdbfe",
            "fontSize": 12,
            "marginTop": 4,
          }
        }
      >
        CSV or PDF
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#e5e7eb",
        "borderTopWidth": 1,
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "#374151",
          "fontSize": 14,
          "fontWeight": "600",
          "marginBottom": 8,
        }
      }
    >
      ðŸ“‹ Supported Formats:
    </Text>
    <Text
      style={
        {
          "color": "#6b7280",
          "fontSize": 12,
          "lineHeight": 16,
          "marginBottom": 4,
        }
      }
    >
      â€¢ CSV files with transaction data
    </Text>
    <Text
      style={
        {
          "color": "#6b7280",
          "fontSize": 12,
          "lineHeight": 16,
          "marginBottom": 4,
        }
      }
    >
      â€¢ PDF bank and credit card statements
    </Text>
    <Text
      style={
        {
          "color": "#6b7280",
          "fontSize": 12,
          "lineHeight": 16,
          "marginBottom": 4,
        }
      }
    >
      â€¢ Maximum file size: 10MB
    </Text>
  </View>
</View>
`;
